packages/xsnap/src/avaXS.js

```diff
@@ -22,10 +21,9 @@ const SESboot = `../dist/bundle-ses-boot-debug.umd.js`;
 const avaAssert = `./avaAssertXS.js`;
 const avaHandler = `./avaHandler.js`;
 
-const importMetaUrl = `file://${__filename}`;
 /** @type { (ref: string, readFile: typeof import('fs').promises.readFile ) => Promise<string> } */
 const asset = (ref, readFile) =>
-  readFile(new URL(ref, importMetaUrl).pathname, 'utf8');
+  readFile(new URL(ref, import.meta.url).pathname, 'utf8');
```

Comment:
hmm... `import.meta.url` is just as much a source of ambient authority as `__filename`, isn't it?

Reply:
Yeah, and the compartment mapper does attenuate that, to the extent that the URL reveals your location within the archive, which includes your own package name, your own module path in that package, and a number of your package’s sequence within the archive. I’m hoping that’s enough attenuation. If it’s not, the next obvious thing would be to change it to a CAS URL, with the hash of the package instead of the the name and sequence number.